The FFI approach is potentially risky

The way to implement FFI is too tricky.

```moonbit
pub extern "c" fn fclose(stream : Ptr[FILE]) -> Int = "fclose"
```
The above code generates the following code

```c
// freopen.c
int32_t fclose(void* $0);
```

But it would actually conflict with the original definition, if include `stdio.h`
```c
fclose.c:11:5: error: conflicting types for ‘fclose’; have ‘int(void *)’
   11 | int fclose(void*);
      |     ^~~~~~
In file included from fclose.c:5:
/usr/include/stdio.h:178:12: note: previous declaration of ‘fclose’ with type ‘int(FILE *)’
  178 | extern int fclose (FILE *__stream);
```

```c
// stdio.h
extern int fclose (FILE *__stream);
```

--- a/src/c.c
+++ b/src/c.c
@@ -38,12 +38,23 @@ FILE *moonbit_c_get_stdout() { return stdout; }
 FILE *moonbit_c_get_stderr() { return stderr; }
 int moonbit_c_get_eof() { return EOF; }
 int moonbit_c_puts(const char *str) { return puts(str); }
+char *moonbit_c_tmpnam(char *str) { return tmpnam(str); }
 int moonbit_c_get_l_tmpnam() { return L_tmpnam; }
 char *moonbit_c_strerror(int errnum) { return strerror(errnum); }
 FILE *moonbit_c_fopen(const char *filename, const char *mode) {
   return fopen(filename, mode);
 }
+int moonbit_c_remove(const char *filename) { return remove(filename); }
+FILE *moonbit_c_freopen(const char *filename, const char *mode, FILE *stream) {
+  return freopen(filename, mode, stream);
+}
+int moonbit_c_ftell(FILE *stream) { return ftell(stream); }
+int moonbit_c_feof(FILE *stream) { return feof(stream); }
+int moonbit_c_fputs(const char *str, FILE *stream) {
+  return fputs(str, stream);
+}
 int moonbit_c_fclose(FILE *stream) { return fclose(stream); }
+void moonbit_c_perror(const char *str) { perror(str); }
 
 void moonbit_c_exit(int status) { exit(status); }
 int moonbit_c_get_exit_success() { return EXIT_SUCCESS; }

--- a/src/stdio.mbt
+++ b/src/stdio.mbt
@@ -12,13 +12,13 @@ pub extern "c" fn freopen(
   filename : Ptr[Byte],
   mode : Ptr[Byte],
   stream : Ptr[FILE]
-) -> Ptr[FILE] = "freopen"
+) -> Ptr[FILE] = "moonbit_c_freopen"
 
 ///|
 pub extern "c" fn fclose(stream : Ptr[FILE]) -> Int = "moonbit_c_fclose"
 
 ///|
-pub extern "c" fn remove(pathname : Ptr[Byte]) -> Int = "remove"
+pub extern "c" fn remove(pathname : Ptr[Byte]) -> Int = "moonbit_c_remove"
 
 ///|
 extern "c" fn get_stdin() -> Ptr[FILE] = "moonbit_c_get_stdin"
@@ -39,16 +39,16 @@ extern "c" fn get_stderr() -> Ptr[FILE] = "moonit_c_get_stderr"
 pub let stderr : Ptr[FILE] = get_stderr()
 
 ///|
-pub extern "c" fn ftell(stream : Ptr[FILE]) -> Int = "ftell"
+pub extern "c" fn ftell(stream : Ptr[FILE]) -> Int = "moonbit_c_ftell"
 
 ///|
-pub extern "c" fn feof(stream : Ptr[FILE]) -> Int = "feof"
+pub extern "c" fn feof(stream : Ptr[FILE]) -> Int = "moonbit_c_feof"
 
 ///|
-pub extern "c" fn fputs(str : Ptr[Byte], stream : Ptr[FILE]) -> Int = "fputs"
+pub extern "c" fn fputs(str : Ptr[Byte], stream : Ptr[FILE]) -> Int = "moonbit_c_fputs"
 
 ///|
-pub extern "c" fn perror(s : Ptr[Byte]) -> Unit = "perror"
+pub extern "c" fn perror(s : Ptr[Byte]) -> Unit = "moonbit_c_perror"
 
 ///|
 pub extern "c" fn get_eof() -> Int = "moonbit_c_get_eof"
@@ -57,7 +57,7 @@ pub extern "c" fn get_eof() -> Int = "moonbit_c_get_eof"
 pub let eof : Int = get_eof()
 
 ///|
-pub extern "c" fn tmpnam(filename : Ptr[Byte]) -> Ptr[Byte] = "tmpnam"
+pub extern "c" fn tmpnam(filename : Ptr[Byte]) -> Ptr[Byte] = "moonbit_c_tmpnam"
 
 ///|
 extern "c" fn get_l_tmpnam() -> UInt64 = "moonbit_c_get_l_tmpnam"

